neon_data_models/models/base/contexts.py

```diff
@@ -85,13 +89,61 @@ def __init__(self, **kwargs):
     transform_utterance: Optional[timedelta] = None
     wait_in_queue: Optional[timedelta] = None
 
+    @model_validator(mode='before')
+    @classmethod
+    def validate_datetime_fields(cls, values):
+        """
+        Validate datetime fields to parse epoch timestamps into datetime objects.
+        Dynamically identifies fields with datetime type annotations.
+        """
+        # Find all fields that have datetime type annotation
+        datetime_fields = [
+            field_name for field_name, field_info in cls.model_fields.items()
+            if field_info.annotation == datetime or (
+                hasattr(field_info.annotation, "__origin__") and 
+                field_info.annotation.__origin__ is Optional and 
+                datetime in field_info.annotation.__args__
+            )
+        ]
+        
+        for field in datetime_fields:
+            if field in values and values[field] is not None:
+                if isinstance(values[field], (int, float)):
+                    # Convert epoch timestamp to datetime with UTC timezone
+                    values[field] = datetime.fromtimestamp(values[field], tz=timezone.utc)
+                elif not isinstance(values[field], datetime):
+                    # If it's neither a timestamp nor a datetime, raise an error
+                    raise ValueError(f"Field '{field}' must be a timestamp or datetime object")
+        
+        return values
+
 
 class KlatContext(BaseModel):
-    sid: str
-    cid: str
-    title: Optional[str] = ""
+    sid: str = Field(default="", description="Klat Shout ID")
+    cid: str = Field(default="", description="Klat Conversation ID")
+    title: Optional[str] = Field(default="",
+                                  description="Klat Conversation Title")
 
+    @model_validator(mode='before')
+    @classmethod
+    def validate_inputs(cls, values):
+        """
+        Validate KlatContext inputs to normalize messageID to sid.
+        """
+        if "sid" not in values and "messageID" in values:
+            values["sid"] = values.get("messageID")
+        return values
 
 class MQContext(BaseModel):
     routing_key: Optional[str] = None
     message_id: str
+
+    @model_validator(mode='before')
+    @classmethod
+    def validate_inputs(cls, values):
+        """
+        Validate MQContext inputs to normalize messageID to message_id.
+        """
+        if "message_id" not in values:
+            values["message_id"] = values.get("messageID")
```

Comment:
could be simplified to `values.setdefault("message_id", values.get("messageID"))`